src/lib/signerVerifier.ts

```diff
@@ -93,7 +94,7 @@ class Ed25519Verifier {
  *    or
  * @param options.signer An object implementing a `sign(CvcMerkleProof)` method
  */
-const signer = async (options: SignerOptions) => {
+const signer = async (options: SignerOptions, didResolver: IDiDResolver) => {
```

Comment:
100% agreed... I'll do that as part of ripping out the proof mechanism which will result in breaking changes (this PR is fully backwards compatible)

Reply:
Oh yes... the problem is that you could have a controller relationship, so it would resolve the doc... check if the VM is a controller of that doc.. then resolve the parent doc, etc